/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

package noisicaa.pb;

message TransferFunctionSpec {
  required float input_min = 1;
  required float input_max = 2;
  required float output_min = 3;
  required float output_max = 4;

  message Fixed {
    required float value = 1;
  }

  message Linear {
    required float left_value = 1;
    required float right_value = 2;
  }

  message Gamma {
    required float value = 1;
  }

  oneof type {
    Fixed fixed = 5;
    Linear linear = 6;
    Gamma gamma = 7;
  }
}
